Kickstarter-iOS/Features/ReportProject/ReportProjectFormView.swift

```diff
@@ -8,45 +8,54 @@ enum ReportFormFocusField {
 
 @available(iOS 15.0, *)
 struct ReportProjectFormView: View {
-  let projectID: String
-  let projectURL: String
-  let projectFlaggingKind: GraphAPI.FlaggingKind
   @Binding var popToRoot: Bool
 
   @SwiftUI.Environment(\.dismiss) private var dismiss
-  @ObservedObject private var viewModel = ReportProjectFormViewModel()
+  @ObservedObject private var viewModel: ReportProjectFormViewModel
 
-  @State private var retrievedEmail = ""
-  @State private var details: String = ""
-  @State private var saveEnabled: Bool = false
-  @State private var saveTriggered: Bool = false
   @State private var showLoading: Bool = false
-  @State private var showBannerMessage = false
-  @State private var submitSuccess = false
-  @State private var bannerMessage: MessageBannerViewViewModel?
   @FocusState private var focusField: ReportFormFocusField?
 
+  init(projectID: String,
+       projectURL: String,
+       projectFlaggingKind: GraphAPI.FlaggingKind,
+       popToRoot: Binding<Bool>) {
+    self._popToRoot = popToRoot
+
+    self.viewModel = ReportProjectFormViewModel(
+      projectID: projectID,
+      projectURL: projectURL,
+      projectFlaggingKind: projectFlaggingKind
+    )
+  }
+
   var body: some View {
     GeometryReader { proxy in
       Form {
-        if !retrievedEmail.isEmpty {
-          SwiftUI.Section(Strings.Email()) {
+        SwiftUI.Section(Strings.Email()) {
+          if let retrievedEmail = viewModel.retrievedEmail, !retrievedEmail.isEmpty {
             Text(retrievedEmail)
               .font(Font(UIFont.ksr_body()))
               .foregroundColor(Color(.ksr_support_400))
               .disabled(true)
+          } else {
+            Text(Strings.Loading())
+              .font(Font(UIFont.ksr_body()))
+              .foregroundColor(Color(.ksr_support_400))
+              .italic()
+              .disabled(true)
           }
         }
 
         SwiftUI.Section(Strings.Project_url()) {
-          Text(projectURL)
+          Text(viewModel.projectURL)
             .font(Font(UIFont.ksr_body()))
             .foregroundColor(Color(.ksr_support_400))
             .disabled(true)
         }
 
         SwiftUI.Section {
-          TextEditor(text: $details)
+          TextEditor(text: $viewModel.detailsText)
```

Comment:
Here's an example of a two-way binding - `TextEditor` can mutate `detailsText`, which will update that property in the viewmodel

Reply:
I could see us passing state back to the viewmodel with an input method like `didSubmitReport(withDetailsText detailsText: String)`. But this also raises another question, which is whether or not it would work with combine and `buttonEnabled`.

Based on what I understand right now - a `@State` variable on a `View` is not a publisher. Right now we're doing this in the viewmodel:
```
    self.$detailsText
      .map { !$0.isEmpty }
      .assign(to: &$saveButtonEnabled)
```
but if `detailsText` is a `@State` and not `@Published`, this pattern doesn't work.

Playing with how this is organized seems like a really good thing to mess around with during a pairing session, it'd be interesting to see what you mean and figure out how else it might work.